### PR TITLE
drivers device interface improvements

### PR DESCRIPTION
**Pull request by dagar**
This can wait until after v1.7.0.

Next step is to split CDev out of Device. Device is the base class for a physical device (eg i2c or spi sensor) with a bus, address, and unique id. There are many CDevs in the system (like all uORB topics) that aren't physical devices.

Additionally there are many device interfaces throughout common drivers like mpu6000, mpu9250, etc that don't need to be CDevs.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@bkueng Could you please review?

**Comment by dagar** [comment]
This might make more sense with a couple diagrams, but basically I'm working towards splitting out CDev from the Device (I2C/SPI) hierarchy.

File: src/modules/uORB/uORBDevices.hpp [hunk 1]
```diff
@@ -177,7 +177,7 @@ class uORB::DeviceNode : public device::CDev
 	 * @param reset if true, reset statistics afterwards
```
Comment (bkueng) [inline src/modules/uORB/uORBDevices.hpp]: This comment needs to be updated (below too)

File: src/drivers/bmi160/bmi160_gyro.cpp [hunk 2]
```diff
@@ -51,13 +51,5 @@ BMI160_gyro::read(struct file *filp, char *buffer, size_t buflen)
 int
 BMI160_gyro::ioctl(struct file *filp, int cmd, unsigned long arg)
 {
-
-	switch (cmd) {
-	case DEVIOCGDEVICEID:
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/drivers/bmi160/bmi160_gyro.cpp]: Yes, part of what this PR needs is thorough checking of device ids.

**Comment by LorenzMeier** [comment]
@dagar what du needed to get the WIP flag removed?

File: src/drivers/device/Device.hpp [hunk 3]
```diff
@@ -104,22 +104,11 @@ class __EXPORT Device
...
-	{
-		switch (operation) {
-		case DEVIOCGDEVICEID:
-			return get_device_id();
-		}
+	virtual int reset() { return PX4_OK; }
 
-		return -ENODEV;
-	}
+	virtual int test(unsigned arg1 = 0, unsigned arg2 = 0) { return PX4_ERROR; }
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/drivers/device/Device.hpp]: It's a helper used in the px4io driver to get away from extra IOCTLs. Let me see if there's a good generalization.

**Pull request closed without merging**